concent-builder/common_tasks/postrequire-configure-tasks.yml

```diff
@@ -1,14 +1,39 @@
 - name:    Install dependencies from testing
   apt:  name={{ item }} default_release=testing
   with_items:
-    - python{{ python_version }}
     - python3-pip
-    - python{{ python_version }}-dev
     # Packages like python3-apt contain files with paths that contain specific Python version (e.g. 3.6).
     # We need to update them to make sure the paths match the interpreter version we have just installed.
     - python3-apt
     - virtualenv
 
+- name:   Clone pyenv repository
+  git:
+    repo:    "{{ pyenv_repository }}"
+    dest:    "{{ pyenv_dir }}"
+    version: "{{ pyenv_version }}"
+    clone:   yes
+    update:  yes
+
+- name:    Install dependencies from pyenv
+  apt:  name={{ item }}
+  with_items:
+    - make
+    - zlib1g-dev
+    - libbz2-dev
+    - libreadline-dev
+    - libsqlite3-dev
+    - llvm
+    - libncurses5-dev
+    - libncursesw5-dev
+    - xz-utils
+    - tk-dev
+
+- name:   Install python3.6 package
+  shell: "{{ pyenv_dir }}/bin/pyenv install {{ specific_python_version_for_penv }}"
+  environment:
+    PYENV_ROOT: "{{ pyenv_dir }}"
```

Comment:
This fails if you run it twice (e.g. by running `vagrant provision` to update the machine.

```
TASK [Install python3.6 package] ***********************************************
fatal: [concent-vm]: FAILED! => {
    "changed": true,
    "cmd": "/opt/pyenv/bin/pyenv install 3.6.8",
    "delta": "0:00:00.017523",
    "end": "2019-02-04 15:52:08.050721",
    "rc": 1,
    "start": "2019-02-04 15:52:08.033198"
}

STDERR:

pyenv: /opt/pyenv/versions/3.6.8 already exists


MSG:

non-zero return code
```

And anyway, Ansible tasks are supposed to be idempotent. It should not run if Python is already installed. You could use `changed_when` for that purpose. See [Error Handling In Playbooks > Overriding The Changed Result](https://docs.ansible.com/ansible/latest/user_guide/playbooks_error_handling.html#overriding-the-changed-result).

Reply:
Alternatively, you could tell `pyenv` to do nothing if the expected version is installed:
``` bash
pyenv install 3.6.8 --skip-existing
```